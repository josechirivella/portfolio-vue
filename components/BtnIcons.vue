<template>
  <a v-if="props?.link && props?.icon" :href="props.link" target="_blank">
    <div class="icon-wrapper">
      <Icon :name="props.icon" color="white" />
    </div>
  </a>
</template>

<script setup lang="ts">
const props = defineProps({
  link: { type: String, required: true },
  icon: { type: String, required: true },
});
</script>

<style lang="scss" scoped>
.icon-wrapper {
  @apply flex items-center justify-center px-4 text-4xl;
}
</style>
