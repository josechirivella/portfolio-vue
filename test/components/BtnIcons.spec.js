import { shallowMount } from '@vue/test-utils';
import BtnIcons from '~/components/BtnIcons';

describe('BtnIcons', () => {
  let wrapper;
  let link;
  let icon;
  beforeAll(() => {
    icon = 'times';
    link = 'https://www.google.com';
    wrapper = shallowMount(BtnIcons, {
      props: {
        link,
        icon,
      },
    });
  });

  test('should render component', () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  test('should render component with link', () => {
    link = 'https://www.facebook.com';
    icon = 'times';
    wrapper.setProps({ link });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
