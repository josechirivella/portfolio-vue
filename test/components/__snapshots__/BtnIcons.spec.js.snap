// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BtnIcons > should render component 1`] = `
<a
  data-v-4943a209=""
  href="https://www.google.com"
  target="_blank"
>
  <div
    class="icon-wrapper"
    data-v-4943a209=""
  >
    <nuxt-icon-stub
      color="white"
      data-v-4943a209=""
      name="times"
    />
  </div>
</a>
`;

exports[`BtnIcons > should render component with link 1`] = `
<a
  data-v-4943a209=""
  href="https://www.facebook.com"
  target="_blank"
>
  <div
    class="icon-wrapper"
    data-v-4943a209=""
  >
    <nuxt-icon-stub
      color="white"
      data-v-4943a209=""
      name="times"
    />
  </div>
</a>
`;
